FreeRTOS_DNS.c

```diff
@@ -830,9 +874,25 @@
                      pxIterator != ( const ListItem_t * ) xEnd;
                      pxIterator = ( const ListItem_t * ) listGET_NEXT( pxIterator ) )
                 {
-                    if( listGET_LIST_ITEM_VALUE( pxIterator ) == uxIdentifier )
+                    BaseType_t xMatching;
+                    DNSCallback_t * pxCallback = ipCAST_PTR_TO_TYPE_PTR( DNSCallback_t, listGET_LIST_ITEM_OWNER( pxIterator ) );
+                    #if ( ipconfigUSE_MDNS == 1 )
+                        /* mDNS port 5353. */
+                        if( pxSet->usPortNumber == FreeRTOS_htons( ipMDNS_PORT ) )
+                        {
+                            /* In mDNS, the query ID field is ignored and the
+                             * hostname will be compared with outstanding requests. */
+
+                            xMatching = ( strcasecmp( pxCallback->pcName, pxSet->pcName ) == 0 ) ? pdTRUE : pdFALSE;
```

Comment:
I spend a lot of time on this statement, thinking of how to avoid the use of a `strcmp()` function. I wrote my own version `string_ignore_case_compare()`, but in fact that would only mimic the behaviour of `strcasecmp()`.
~~~c
    strcasecmp( pxCallback->pcName, pxSet->pcName )
    strcmp( pxCallback->pcName, pxSet->pcName )
    strncmp( pxCallback->pcName, pxSet->pcName, uxLength )
    memcmp( pxCallback->pcName, pxSet->pcName, uxLength )
~~~
Now I removed all deviations from `coverity_misra.config` and checked the above code: there is no warning at allexcept for `memcmp()`. Coverity will say that:
"[Required] MISRA C-2012 Rule 21.14 violation: Null terminated strings "pxCallback->pcName" and "pxSet->pcName" are compared in function "memcmp".". Which sounds good.

But we look at this again in PR #358.

Reply:
Makes sense, thank you.